pkg/controller/controller.go

```diff
@@ -40,51 +56,46 @@ func normalize(jaeger *v1alpha1.Jaeger) {
 	// normalize the storage type
 	if jaeger.Spec.Storage.Type == "" {
 		logrus.Infof("Storage type wasn't provided for the Jaeger instance '%v'. Falling back to 'memory'", jaeger.Name)
-		jaeger.Spec.Storage.Type = "memory"
+		jaeger.Spec.Storage.Type = memoryStorageType
 	}
 
 	if unknownStorage(jaeger.Spec.Storage.Type) {
 		logrus.Infof(
 			"The provided storage type for the Jaeger instance '%v' is unknown ('%v'). Falling back to 'memory'. Known options: %v",
 			jaeger.Name,
 			jaeger.Spec.Storage.Type,
-			knownStorages(),
+			knownStorages,
 		)
-		jaeger.Spec.Storage.Type = "memory"
+		jaeger.Spec.Storage.Type = memoryStorageType
 	}
 
 	// normalize the deployment strategy
-	if strings.ToLower(jaeger.Spec.Strategy) != "production" {
-		jaeger.Spec.Strategy = "all-in-one"
+	if strings.ToLower(jaeger.Spec.Strategy) != productionStrategy {
+		jaeger.Spec.Strategy = allInOneStrategy
 	}
 
 	// check for incompatible options
 	// if the storage is `memory`, then the only possible strategy is `all-in-one`
-	if strings.ToLower(jaeger.Spec.Storage.Type) == "memory" && strings.ToLower(jaeger.Spec.Strategy) != "all-in-one" {
+	if strings.ToLower(jaeger.Spec.Storage.Type) == memoryStorageType && strings.ToLower(jaeger.Spec.Strategy) != allInOneStrategy {
 		logrus.Warnf(
 			"No suitable storage was provided for the Jaeger instance '%v'. Falling back to all-in-one. Storage type: '%v'",
 			jaeger.Name,
 			jaeger.Spec.Storage.Type,
 		)
-		jaeger.Spec.Strategy = "all-in-one"
+		jaeger.Spec.Strategy = allInOneStrategy
 	}
 }
 
 func unknownStorage(typ string) bool {
-	for _, k := range knownStorages() {
-		if strings.ToLower(typ) == k {
+
+	for _, v := range knownStorages {
+
+		if v == strings.ToLower(typ) {
 			return false
 		}
+
 	}
 
 	return true
-}
 
-func knownStorages() []string {
```

Comment:
pity that there are no constant arrays in go. I think I like more this approach than defining it as private array variable. I am not sure though how other golangers write this.